admin-dev/themes/new-theme/js/components/event-emitter.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import EventEmitter from 'events';
+
+/**
+ * We instanciate one EventEmitter (accessible from window) so that every components
+ * register/dispatch on the same one and can communicate with each other.
+ */
+if (!window.PrestaShopEventEmitter) {
+  window.PrestaShopEventEmitter = new EventEmitter();
+}
+
+export default window.PrestaShopEventEmitter;
```

Comment:
```js
import EventEmitter from 'events';

export const PrestaShopEventEmitter = new EventEmitter();
```
Maybe try this

Reply:
Thanks @PierreRambaud so much better!
I preferred keeping `EventEmitter` so as the const name.